### PR TITLE
Fix time_comparison for dates

### PR DESCRIPTION
**Pull request by bbenno**
Assume an ActiveRecord with a release date

```
has_event :release, field_type: :date, strategy: :time_comparison
```

and let's further that the release date is `Date.today`.
Previously calling `#released?` will return `false` due to [checking with `#past?`][1]. Although, `true` might have been expected in this situation due to considering the release date as "the first date being released".
The new behavior returns `true` for exactly this edge case of date-events with time_comparison strategy by using `! #future?` instead of `#past?`.

[1]: https://github.com/pienkowb/active_record-events/blob/2814f2d3828ab69b4cb22890a50d0a77b21ff7cd/lib/active_record/events/method_factory.rb#L42

EDIT: an alternative approach (in case the previous behavior is intended) I favor a new strategy with the described behavior.

### CODE REVIEW

**Comment by pienkowb** [comment]
Thanks for the pull request! I think the correct behavior is to include the current date, so we should update the logic instead of adding a new strategy.

Could you also add a test example that would verify this edge case (in [this section](https://github.com/pienkowb/active_record-events/blob/develop/spec/active_record/events_spec.rb#L75))?

By the way, this seems to be in line with the logic we have in scope methods ([here](https://github.com/pienkowb/active_record-events/blob/8d135bb0187a97c50b033314f0302600f89047ca/lib/active_record/events/method_factory.rb#L84) and below).

**Comment by bbenno** [comment]
Hi @pienkowb, thanks for sharing your thoughts. I am glad to see you in favor of this change.

I extended the test class `Task` and added the requested tests.

File: spec/active_record/events_spec.rb [hunk 1]
```diff
@@ -116,5 +116,22 @@
       task.update_columns(expired_at: nil)
       expect(Task.not_expired).to include(task)
     end
+
+    describe 'for date fields' do
+      it 'consider today\'s event over' do
```
Comment (pienkowb) [inline spec/active_record/events_spec.rb:121]: Let's use double quotes here (and below) to avoid escaping characters.

File: spec/active_record/events_spec.rb [hunk 2]
```diff
@@ -116,5 +116,22 @@
...
+        task.update_columns(notified_on: Date.today)
+        expect(task.notified?).to eq(true)
```
Comment (pienkowb) [inline spec/active_record/events_spec.rb:123]: Could you also check if the scope returns the record? Similar to: https://github.com/pienkowb/active_record-events/blob/d727d184ff9eec0a28f92a24ef4157ebd2724695/spec/active_record/events_spec.rb#L103

**Review by pienkowb — COMMENTED** [review]
I've added two minor suggestions but overall the PR looks good 👍

File: spec/active_record/events_spec.rb (hunk 1 above)
Comment (bbenno) [reply spec/active_record/events_spec.rb:121]: ah, of course :)

File: spec/active_record/events_spec.rb (hunk 2 above)
Comment (bbenno) [reply spec/active_record/events_spec.rb:123]: nice idea. done.

**Review by pienkowb — APPROVED** [review]
@bbenno Looks like one of the CI jobs has failed: https://github.com/pienkowb/active_record-events/actions/runs/4392746160/jobs/7693008520

I think it's unrelated to this PR though. Most likely we need to compare against `Date.current` instead of `Time.current` [here](https://github.com/pienkowb/active_record-events/blob/2814f2d3828ab69b4cb22890a50d0a77b21ff7cd/lib/active_record/events/method_factory.rb#L84) and [here](https://github.com/pienkowb/active_record-events/blob/2814f2d3828ab69b4cb22890a50d0a77b21ff7cd/lib/active_record/events/method_factory.rb#L96
). The error shows up only when there's a significant timezone difference (in specs a timezone is randomly set by the Zonebie gem).

I'll handle this issue in a separate PR. Thanks for the contribution! 👍

**Comment by pienkowb** [comment]
@bbenno I've released these changes in version [4.1.2](https://rubygems.org/gems/active_record-events/versions/4.1.2).

**Comment by bbenno** [comment]
Hey, thanks for the heads-up. I am going to update my affected projects and appreciate the refactoring you did.
